mesop/protos/ui.proto

```diff
@@ -145,13 +146,24 @@ message ContextLine {
 
 message RenderEvent {
     optional Component root_component = 1;
-    optional ComponentDiff component_diff = 6;
-    optional string title = 5;
-    optional States states = 2;
-    repeated Command commands = 3;
+    optional ComponentDiff component_diff = 2;
+    optional string title = 3;
+    repeated Command commands = 4;
 
     // Only sent in editor mode:
-    repeated ComponentConfig component_configs = 4;
+    repeated ComponentConfig component_configs = 5;
+}
+
+// UI response event for updating state.
+message UpdateStateEvent {
+    optional States states = 1;
```

Comment:
Instead of using an enum, you could model this as:

oneof state_type {
   States full_states = 1
   States state_diffs = 2;
}

I think it's a little clearer, but if you prefer enum, that's OK since this in an implementation detail.

Reply:
Funny thing about this is that earlier this week, someone suggested I change my oneof to use enums since all the oneofs were the same data type. I didn't quite buy the reasoning. Anyway updated to oneof.

Reply:
lol